integration-tests/test_apalache_integration.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+
+import pytest
+import socket
+import time
+
+from subprocess import Popen
+from chai import Chai
+
+
+# Utility function
+def wait_for_server(server: str, port: int, timeout: int) -> bool:
+    """Ping a server
+
+    Adapated from https://stackoverflow.com/a/67217558/1187277
+    """
+    remaining_time = timeout
+    while remaining_time > 0:
+        socket.setdefaulttimeout(remaining_time)
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        try:
+            s.connect((server, port))
+        except TimeoutError:
+            # We ran out of time waiting for a connection on the socket
+            return False
+        except OSError:
+            # We weren't able to connect to the socket, wait a bit, and try again
+            remaining_time -= 1
+            time.sleep(1)
+        else:
+            # We connected!
+            s.close()
+            return True
+    else:
+        # We ran out of remaining_time, so give up
+        return False
+
+
+# Fixture to start and clean up Apalache's Shai server
+#
+# - `autouse=True`:
+#
+#   Ensures that the fixutre is provided (i.e., that the server is started)
+#   for every test.
+#
+# - `scope="module"`:
+#
+#   Specifies that this fixture is created once for every test in the module,
+#   rather than created over and over See
+#   https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session
+@pytest.fixture(autouse=True, scope="module")
```

Comment:
Is the following accurate?
If so, I think it provides a clearer description.
```suggestion
#   Specifies that this fixture is created only once for all tests in the module,
#   rather than created once per test. See
#   https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session
@pytest.fixture(autouse=True, scope="module")
```

Reply:
Yep, that’s correct and better. Thanks!